.github/workflows/ssh_debug.yaml

```diff
@@ -0,0 +1,47 @@
+name: Debug passwordless `ssh localhost`
+
+on: []
+# on: [pull_request]  # Uncomment to enable
+
+jobs:
+  test:
+    runs-on: ${{ matrix.os }}
+    strategy:
+      fail-fast: false
+      matrix:
+        os:
+          - ubuntu-latest
+          - macos-latest
+          # - windows-latest  # FIXME https://github.com/dask/distributed/issues/4509
+
+    steps:
+      - name: Setup SSH
+        shell: bash -l {0}
+        run: bash continuous_integration/scripts/setup_ssh.sh
+
+      - name: Dump SSH config
+        shell: bash -l {0}
+        run: |
+          ls -ld ~ ~/.ssh ~/.ssh/*
+          for f in ~/.ssh/* /etc/ssh/sshd_config; do
+            echo ==================================
+            echo $f
+            echo ==================================
+            cat $f
+          done
+
+      - name: Test SSH vs. localhost
+        shell: bash -l {0}
+        run: ssh -vvv localhost 'echo hello world'
+
+      - name: Test SSH vs. 127.0.0.1
+        shell: bash -l {0}
+        run: ssh -vvv 127.0.0.1 'echo hello world'
+
+      - name: Test SSH vs. hostname
+        shell: bash -l {0}
+        run: ssh -vvv $(hostname) 'echo hello world'
+
+      - name: Debug with tmate on failure
+        if: ${{ failure() }}
+        uses: mxschmitt/action-tmate@v3
```

Comment:
I think this should get into master; it will save *a lot* of work next time SSH breaks.